engine/src/main/java/org/terasology/rendering/dag/NodeFactory.java

```diff
@@ -30,20 +30,18 @@ public NodeFactory(Context context) {
     }
 
     public <T extends Node> T createInstance(Class<T> type) {
-        // Attempt constructor-based injection first
-        T node = InjectionHelper.createWithConstructorInjection(type, context);
-        // Then fill @In fields
-        InjectionHelper.inject(node, context);
-        node.initialise();
-        return type.cast(node);
+        return createInstance(type, false);
     }
 
-    public <T extends Node> T createInstance(Class<T> type, Object initializationData) {
+    public <T extends Node> T createInstance(Class<T> type, boolean delayInitialization) {
         // Attempt constructor-based injection first
         T node = InjectionHelper.createWithConstructorInjection(type, context);
         // Then fill @In fields
         InjectionHelper.inject(node, context);
-        node.initialise(initializationData);
+        if (!delayInitialization) {
+            node.initialise();
+        }
```

Comment:
This allows instantiation to occurs but initialization can be delayed, usually to the next line in WorldRendererImpl.initRenderGraph(). It kind of works in terms of readability for the time being. I just hope it's not some kind of anti-pattern.